### PR TITLE
feat: `decodeIdToken()`

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- implement `decodeIdToken()`

## Testing

UT

## Reviwers

@logto-io/eng

### CODE REVIEW

File: Sources/Logto/Extensions/Collection.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+//
```
Comment (simeng-li) [inline Sources/Logto/Extensions/Collection.swift:1]: 🤣  multi-line comment?

File: Sources/Logto/LogtoUtilities.swift [hunk 2]
```diff
@@ -8,6 +8,13 @@
 import CommonCrypto
 import Foundation
 
+public enum LogtoErrors {
```
Comment (simeng-li) [inline Sources/Logto/LogtoUtilities.swift:11]: Do we need to extract the LogtoErrors into an independent file?

**Review by wangsijie — APPROVED** [review]
LGTM

File: Sources/Logto/LogtoUtilities.swift [hunk 3]
```diff
@@ -25,4 +32,24 @@ public enum LogtoUtilities {
         }
         return Data(hash).toUrlSafeBase64String()
     }
+
+    /// Decode ID Token claims WITHOUT validation.
+    /// - Parameter token: The JWT to decode
+    /// - Returns: A set of ID T	oken claims
```
Comment (simeng-li) [inline Sources/Logto/LogtoUtilities.swift:38]: T      oken typo

File: Sources/Logto/LogtoUtilities.swift (hunk 2 above)
Comment (gao-sun) [reply Sources/Logto/LogtoUtilities.swift:11]: yes, i did this in the next commit, but reverted 😂

File: Sources/Logto/Extensions/Collection.swift (hunk 1 above)
Comment (gao-sun) [reply Sources/Logto/Extensions/Collection.swift:1]: this is generated by Xcode, looks like a convention of swift projects

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**